service/web/publisherserver.go

```diff
@@ -34,11 +35,10 @@ func (s publisherServer) Publish(ctx context.Context, req *metrov1.PublishReques
 		"topic": req.Topic,
 	})
 	defer span.Finish()
-
-	if ok, err := s.topicCore.ExistsWithName(ctx, req.Topic); err != nil {
-		return nil, merror.ToGRPCError(err)
-	} else if !ok {
-		return nil, merror.New(merror.NotFound, "topic not found").ToGRPCError()
+	if tasks.CheckIfTopicExists(ctx, req.Topic) {
+		logger.Ctx(ctx).Infow("Topic exists inside the cache..", "req", req.Topic)
+	} else {
+		return nil, merror.New(merror.NotFound, "Topic not found inside the cache..").ToGRPCError()
```

Comment:
This is not a pre-existing GRPC error, can we avoid this and stick to the existing error message?